### ISSUE TITLE
Forge 1.14 support

### ISSUE THREAD

**Issue by astei** [VelocityPowered/Velocity#248]

**Comment by astei** [comment]
There has been a general lack of interest in supporting Forge versions past 1.12.2, so I'm taking this off the roadmap. We will still entertain support for Forge 1.13+ in Velocity if there is sufficient commitment to supporting it and the Forge team offers a viable path towards adding proxy support with their new handshaking protocol.

**Comment by phit** [comment]
That's sad to hear, can't really gather interest if there is no support. A lot of people are stuck on older versions with modded right now due to stuff like Sponge still not having updated. 
Are there any open Forge issues since you seem to hint at Forge adding proxy support?

**Comment by XinKe-Conch** [comment]
We really need forge. Please don't give it up. Plz

**Comment by astei** [comment]
@XinKe-Conch Velocity still supports Forge 1.7-1.12.2 and we do not intend to drop support for them any time soon. We are referring to support for Forge 1.14+.

**Comment by XinKe-Conch** [comment]
Yes, although the 1.14 + forge is gradually replaced by fabric, there are still some excellent mod developers using forge for development. I hope you can support it

**Comment by Dec1lent** [comment]
> @XinKe-Conch Velocity still supports Forge 1.7-1.12.2 and we do not intend to drop support for them any time soon. We are referring to support for Forge 1.14+.

Is there a way to support 1.4.7 forge on ProtocolSupportVelocity? Im running multiple modpacks in all different versions (💯

**Comment by astei** [comment]
It is extremely unlikely Velocity will support ancient versions of Minecraft, let alone ancient versions of Forge.

**Comment by astei** [comment]
I wanted to provide a little update on this. There is currently a work-in-progress mod to try and enable this on Forge 1.16+ by reintroducing the old-style handshakes in a newer version of Forge that is in cahoots with the proxy to allow switching over to the login state and redo the handshake. I do not know when this work will be complete but there is already a proof of concept.

**Comment by PurpleIsEverything** [comment]
While this is good news one should remember any such solution will absolutely require a client side mod.

You then have to get modpacks and the like to include that mod otherwise it doesn't help all that much. Also the possibility of such a solution causing a great many mod specific issues is staggering.

This solution mentioned is more functional as a proof of concept to give direction to a proper native Forge solution.

**Comment by PurpleIsEverything** [comment]
I should also add we had discussions with Lex he has been willing to accept a PR containing this functionality for a while he simply was not willing to put the time into the creation himself.

**Comment by Xernium** [comment]
@PurpleIsEverything well dang. I’ll try to come up with something workable in a reasonable amount of time. And I agree, modifying forge itself should solve more issues down the line. I’ve already been working on a few things relating to it so it shouldn’t be too much more work. Stay tuned.

**Comment by PurpleIsEverything** [comment]
We've spent quite a bit of time investigating and attempting to implement similar solutions into Bungee and Velocity 

I'm in both the PaperMC and Velocity discords following the progress of these solutions and we've had discussions with Lex here not too long ago.

I'd be happy to help in the development my discord name is PurpleIsEverything#0001

**Comment by SirOMGitsYOU** [comment]
Hey has there been any progress on a potential solution? I’m looking to upgrade my network for 1.16 but can’t do so without velocity forge support as my network is basically all forge servers

**Comment by Xernium** [comment]
I expect to present a working solution by New Years. I’ll have more time by then, stay tuned.

**Comment by SirOMGitsYOU** [comment]
> I expect to present a working solution by New Years. I’ll have more time by then, stay tuned.

Hey what's the progress on this very eager to update my stuff to 1.16

**Comment by segg21** [comment]
Please I support this, 100%.

**Comment by SirOMGitsYOU** [comment]
Has there been much movement on this now that 1.17 is sorted?

**Comment by plusls** [comment]
any progess?

**Comment by astei** [comment]
Requests to ask for the status on features we may add to Velocity will be ignored. If we haven't provided an update, then there is no progress.

**Comment by danorris709** [comment]
Sorry you're saying it was not accepted? https://github.com/PaperMC/Velocity/pull/690#pullrequestreview-943904486

To be quite frank your entire attitude to this PR has been disgusting from the start @electronicboy. I've seen private comments made by you where you're openly disrespecting the Pixelmon team and all the work we're trying to put in to make a fix for the community. I can't tell if you're actually against Forge support or just mad because someone else did it before you.

Ultimately, this should not come down to "we want it done this way" or "we want it done that way". No one else in the Forge or Velocity community is willing to put a solution forward and there is plenty of support for said solution. Closing this PR does not resolve the fact that there is a demand for this solution. We will continue to maintain a fork of Velocity and we will continue to distribute it to the servers with tens of thousands of players, that **need** this solution. 

Ultimately, this is a complete and utter failure on your part to serve the community. 

In regards to the now locked PR https://github.com/PaperMC/Velocity/pull/690

**Comment by electronicboy** [comment]
I'm saying that the PR was not an acceptable solution for us, as we've generally been saying for months. We want it done a specific way which is acceptable to the wider community, closing the PR does iterate that that is not an acceptable solution for us, and given that it's just a back and forth of people disagreeing on the solution here is a complete waste of time. I literally do not care who comes up with a solution, but we want something that actually will be integrated into forge, not another "just tell your users to install this mod to connect" solution.

I'm not aware of any comments I've made on the pixelmon team, only on the mod itself and my experiences of dealing with it as a host which was from years ago, and entirely irrelevant to that PR and such.
